nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -302,11 +305,22 @@ public Description matchNewClass(NewClassTree tree, VisitorState state) {
     if (!matchWithinClass) {
       return Description.NO_MATCH;
     }
-    final Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(tree);
+    Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(tree);
     if (methodSymbol == null) {
       throw new RuntimeException("not expecting unresolved method here");
     }
     List<? extends ExpressionTree> actualParams = tree.getArguments();
+    if (tree.getClassBody() != null && actualParams.size() > 0) {
+      // passing parameters to constructor of anonymous class
+      // this constructor just invokes the constructor of the superclass, and
+      // in the AST does not have the parameter nullability annotations from the superclass.
+      // so, treat as if the superclass constructor is being invoked directly
+      // see https://github.com/uber/NullAway/issues/102
+      Type supertype = state.getTypes().supertype(methodSymbol.owner.type);
+      Symbol.MethodSymbol superConstructor =
+          findSuperConstructorInType(methodSymbol, supertype, state.getTypes());
+      methodSymbol = superConstructor;
+    }
```

Comment:
Error out if `findSuperConstructorInType` returns `null`? Or keep the original `methodSymbol`? Or assert this is impossible?